transcom/mymove
# B-18033: Add createSitAddressUpdateRequest logic to createNonSitAddressUpdateRequest

## [B-18033](https://www13.v1host.com/USTRANSCOM38/Default.aspx?menu=StoryBoardPage&feat-nav=--m2)

## Summary

This pull request introduces backend changes that includes moving needed functionality from createSitAddressUpdateRequest method to createNonSitAddressUpdateRequest. "createSitAddressUpdateRequest" will soon be deprecated and createNonSitAddressUpdateRequest will be used to make shipment destination address changes and will be renamed to "updateShipmentDestinationAddress". Logic checks such as address change distance calculations, and move status updates were in the createSitAddressUpdateRequest method but not in createNonSitAddressUpdateRequest. These checks were copied over from the serviceItem level to the shipment level by adding them to createNonSitAddressUpdateRequest. "createNonSitAddressUpdateRequest" also implements checks for changes with service area, mileage bracket, and weight bracket. This allows these checks to be ran for address changes on the shipment level to tell whether  the potential address change will cause a cost change. If there is a cost change then the address change requires TOO review.

Is there anything you would like reviewers to give additional scrutiny? None

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. In swagger or postman send a request using the createSITAddressUpdateRequest endpoint.
2. Use a zipcode that is less than 50 miles away
![image](https://github.com/transcom/mymove/assets/136514590/049cb9d8-ff90-458b-9241-50835ac6003b)
3. Navigate to the TOO and notice the Destination Address Review Required indicator
![image](https://github.com/transcom/mymove/assets/136514590/daf760a4-bcac-43a0-9616-25c94649a504)
This indicator is triggered because it requires TOO Approval due to the potential address change resulting in a cost change.

### Backend
